src/com/twitter/intellij/pants/execution/PantsMakeBeforeRun.java

```diff
@@ -285,29 +293,57 @@ public void onTextAvailable(ProcessEvent event, Key outputType) {
     runningPantsProcesses.remove(currentProject, process);
 
     final boolean success = process.exitValue() == 0;
-    // Mark project dirty if compile failed.
-    if (!success) {
-      FileChangeTracker.markDirty(currentProject);
-    }
-    else {
-      FileChangeTracker.addManifestJarIntoSnapshot(currentProject);
+    if (tasks.contains(PantsConstants.PANTS_TASK_COMPILE)) {
+      if (success) {
+        // FIXME: is manifest jar always created? what is that? IS it always, if there's a compile?
+        FileChangeTracker.addManifestJarIntoSnapshot(currentProject);
+      } else {
+        // Mark project dirty if compile failed.
+        FileChangeTracker.markDirty(currentProject);
+      }
+      // Sync files as generated sources may have changed after Pants compile.
+      PantsUtil.synchronizeFiles();
```

Comment:
You are changing the behavior here, now it will only sync if pants run succeeds. We need to sync it regardless.

Reply:
This is outside the `if (success) else` block, so it just runs if one of the tasks is `compile`, which is the same behavior as before since all pants actions call compile as of this commit. I added a blank line before the `// Sync files ...` comment so this is more clear.